### ISSUE TITLE
Range error when a class contains a static final list field

### ISSUE THREAD

**Issue by zoechi** [dart-lang/reflectable#34]
```dart
@PolymerRegister('my-element')
class MyElement extends PolymerElement {
  static const List<String> hardCodedListOfWords = const [
    "apple"];
  MyElement.created() : super.created() {}
}
```

When I rename it to make it private it works fine.
Full project https://github.com/bwu-dart-contributing/polymer-dart-patterns/blob/0.17.0/web/forms/implementing_simple_autocompletion/my_element.dart#L17

>  Exception: Uncaught Error: RangeError (index): Invalid value: Not in range 0..2523, inclusive: -1
Stack Trace:
#0      List.[] (dart:core-patch/growable_array.dart:135)
#1      ClassMirrorImpl.staticMembers (package:reflectable/src/reflectable_transformer_based.dart:352:32)
#2      _setupHostAttributes (package:polymer/src/common/polymer_descriptor.dart:160:18)
#3      createPolymerDescriptor (package:polymer/src/common/polymer_descriptor.dart:32:3)
#4      PolymerRegister.initialize (package:polymer/src/common/polymer_register.dart:19:36)
#5      loadInitializers.<anonymous closure>.<anonymous closure> (package:initialize/src/static_loader.dart:46:32)
#6      _runInitQueue (package:initialize/initialize.dart:35:24)
#7      _runInitQueue.<anonymous closure> (package:initialize/initialize.dart:38:26)
#8      _RootZone.runUnary (dart:async/zone.dart:1165)
#9      _Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:502)
#10     _Future._propagateToListeners (dart:async/future_impl.dart:585)
#11     _Future._completeWithValue (dart:async/future_impl.dart:376)
#12     _Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:430)
#13     _microtaskLoop (dart:async/schedule_microtask.dart:43)
#14     _microtaskLoopEntry (dart:async/schedule_microtask.dart:52)
#15     _ScheduleImmediateHelper._handleMutation (dart:html:42565)

**Comment by jakemac53** [comment]
Interesting were there any errors in the transformer?

**Comment by zoechi** [comment]
No, only in the browser console. 
I run into the same issue here https://github.com/bwu-dart-contributing/polymer-dart-patterns/blob/0.17.0/web/forms/implementing_simple_validation/my_element.dart#L9

**Comment by sigurdm** [comment]
Thanks for the report!

I tried for a while to reproduce but did not manage.
It seems that `implementing_simple_autocompletion`'s entry-point is not added to the entry_points of the reflectable transformer in the `pubspec.yaml`. Do I have to make some modifications from the  0.17 branch of https://github.com/bwu-dart-contributing/polymer-dart-patterns in order to reproduce the error you saw? Do you then run with `pub serve` and `dartium`?

**Comment by zoechi** [comment]
It's the `0.17.0` branch not the `0.17` branch. Sorry for the confusion. There the entry-point is included.

**Comment by eernstg** [comment]
Interesting! ;)

The path PolymerRegister.initialize -> createPolymerDescriptor -> _setupHostAttributes -> ClassMirrorImpl.staticMembers is exactly one of those that I've just tracked down. My next step was asking Jacob MacDonald for more information about the case where he is encountering an empty result from `staticMembers` which might arise in that path. There might be a connection here.

**Comment by sigurdm** [comment]
I managed to reproduce (sorry I somehow got the wrong branch), and will try to work on a fix tomorrow

**Comment by sigurdm** [comment]
This seems to have been fixed by a468609 can you confirm?
